contrib/awslambda/python/src/python/pants/contrib/awslambda/python/tasks/lambdex_run.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import functools
+import os
+import shutil
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.task.task import Task
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import safe_mkdir_for
+from pants.util.fileutil import atomic_copy
+
+from pants.contrib.awslambda.python.targets.python_awslambda import PythonAWSLambda
+from pants.contrib.awslambda.python.tasks.lambdex_prep import LambdexPrep
+
+
+class LambdexRun(Task):
+  """Runs Lambdex to convert pexes to AWS Lambda functions.
+
+  Note that your pex must be built to run on Amazon Linux, e.g., if it contains native code.
+  When deploying the lambda, its handler should be set to `lambdex_handler.handler`, which
+  is a wrapper around the target-specified handler.
+  """
```

Comment:
On the face of it, I would love it if an amazon linux image could be added to our set of travis images before merging this. I don't have a problem with merging this otherwise except that I would feel uncomfortable if the only testing was outside the open-source repo. I think the ability to test aws lambda functions would be a *fantastic* feature to incorporate on its own, and would love to discuss in an issue or something how best to incorporate that kind of testing. I'm mostly concerned about bitrot -- this is a great feature and I'm trying to not block it. Not sure if there is context I'm missing here wrt testing, and not sure how that testing would be accomplished.

Reply:
`moto`, the mock AWS library, does support Lambda APIs, but requires Docker to actually run them. That seems like the right path, but that may be a non-trivial change to our CI, I think. Still, running Amazon Linux in a Docker container is likely easier and less costly than having a travis image just for this one thing.